source/Calamari/ArgoCD/Conventions/UpdateArgoCDAppImagesInstallConvention.cs

```diff
@@ -73,6 +74,7 @@ public void Install(RunningDeployment deployment)
                 var valuesFilesToUpdate = new List<HelmValuesFileImageUpdateTarget>();
                 var applicationFromYaml = argoCdApplicationManifestParser.ParseManifest(application.Manifest);
                 gatewayIds.Add(application.GatewayId);
+                bool containsMultipleSources = applicationFromYaml.Spec.Sources.Count > 1;
```

Comment:
Do we need to show an error here if the annotations/sources are 'illegal'? Eg un-named annotation and more than 1 source? (i.e. its more important that just 'we didn't update your appSource'

Reply:
I will add that (plus validating for unique names in a separate PR)